automated_analysis.py

```diff
@@ -464,5 +459,42 @@ def set_survey_percentages(survey_counts, total_survey_counts):
         fig.update_xaxes(tickangle=-60)
         fig.write_image(f"{automated_analysis_output_dir}/graphs/{plan.raw_field}_by_gender_normalised.png",
                         scale=IMG_SCALE_FACTOR)
+    
+    # Export safe to share raw messages for each episode to share with WUSC
+    log.info("Exporting safe to share raw messages for each episode...")
+    safe_to_share_messages = []  # of dict
+    for plan in PipelineConfiguration.RQA_CODING_PLANS[5:]: #loop through episode 6 -> because previous episodes had been manually processed
+        for cc in plan.coding_configurations:
+            code_to_messages = dict()
+            for code in cc.code_scheme.codes:
+                code_to_messages[code.string_value] = []
+
+            for msg in messages:
+                if not AnalysisUtils.opt_in(msg, CONSENT_WITHDRAWN_KEY, plan):
+                    continue
+
+                for label in msg[cc.coded_field]:
+                    code = cc.code_scheme.get_code_with_code_id(label["CodeID"])
+                    code_to_messages[code.string_value].append(msg[plan.raw_field])
+
+            for code_string_value in code_to_messages:
+                if code_string_value == "DNS": # TODO add DNS control code to core and update here
```

Comment:
I think this just excludes messages which only have a DNS label. Messages which have DNS and another code will still be exported, but only with the code column

Reply:
True, have refactored the code from control code -> meta code because we lose control code label when a normal code is assigned to the same message. Have also refactored the logic to prevent this.

Reply:
Tested again and we don't loose the control code label - updated dns -> control code